packages/core/src/node-execution-context/execute-context.ts

```diff
@@ -0,0 +1,263 @@
+import type {
+	CallbackManager,
+	CloseFunction,
+	ExecutionBaseError,
+	IExecuteData,
+	IExecuteFunctions,
+	IExecuteResponsePromiseData,
+	IGetNodeParameterOptions,
+	INode,
+	INodeExecutionData,
+	IRunExecutionData,
+	ITaskDataConnections,
+	ITaskMetadata,
+	IWorkflowExecuteAdditionalData,
+	NodeConnectionType,
+	Workflow,
+	WorkflowExecuteMode,
+} from 'n8n-workflow';
+import { ApplicationError, createDeferredPromise } from 'n8n-workflow';
+
+import { createAgentStartJob } from '@/Agent';
+// eslint-disable-next-line import/no-cycle
+import {
+	returnJsonArray,
+	copyInputItems,
+	normalizeItems,
+	constructExecutionMetaData,
+	getInputConnectionData,
+	addExecutionDataFunctions,
+	assertBinaryData,
+	getBinaryDataBuffer,
+	copyBinaryFile,
+	getRequestHelperFunctions,
+	getBinaryHelperFunctions,
+	getSSHTunnelFunctions,
+	getFileSystemHelperFunctions,
+	getCheckProcessedHelperFunctions,
+} from '@/NodeExecuteFunctions';
+
+import { BaseExecuteContext } from './base-execute-context';
+
+export class ExecuteContext extends BaseExecuteContext implements IExecuteFunctions {
```

Comment:
This is somewhat confusing. We have `ExecuteContext` that extends `BaseExecuteContext` that extends `NodeExecutionContext`. Could we have some more meaningful names for these? I don't have enough context to understand the difference between them

Reply:
I want to rename all these, and add extensive docs on each of these classes, and all of the public methods. But I want to do that in https://linear.app/n8n/issue/CAT-280/add-inline-docs-for-all-the-context-objects